### PR TITLE
NIAD-1152: Add the message type to GP Outbound queue items

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
## Description

A third header, `MessageType`, is added to the JMS
item for the GP outbound queue with the original
from the edifact message header, e.g. "NHS003"

## Jira Ticket

https://gpitbjss.atlassian.net/browse/NIAD-1152

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [x] Acceptance Criteria met
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [x] If your pull request depends on any other, please link them in the description
- [x] main branch is passing/green on CI
- [x] Add/update any relevant documentation

## Evidence

image

### CODE REVIEW

**Review by saileshp-kainos — DISMISSED** [review]
LGTM

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/MessageHeader.java [hunk 1]
```diff
@@ -37,14 +41,18 @@ public void validate() {
...
     public static MessageHeader fromString(final String edifactString) {
         if (!edifactString.startsWith(KEY)) {
             throw new IllegalArgumentException("Can't create " + MessageHeader.class.getSimpleName()
                 + " from " + edifactString);
         }
-        final String[] split = Split.byPlus(edifactString);
-        return new MessageHeader(Long.valueOf(split[1]));
+        final String[] splitByPlus = Split.byPlus(edifactString);
+        final String[] splitByColon = Split.byColon(splitByPlus[2]);
+        return new MessageHeader(Long.valueOf(splitByPlus[1]), splitByColon[INDEX_MESSAGE_TYPE]);
```
Comment (graham-andrew) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/MessageHeader.java:56]: Should we add another constant field for the index of the sequence number, to make it obvious what we're trying to extract here?
```suggestion
        return new MessageHeader(Long.valueOf(splitByPlus[INDEX_SEQUENCE_NUMBER]), splitByColon[INDEX_MESSAGE_TYPE]);
```

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/outbound/queue/GpOutboundQueueServiceTest.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
 class GpOutboundQueueServiceTest {
 
     private static final String CONSERVATION_ID = "ABC123";
+    private static final String MESSAGE_TYPE = "NHS003";
```
[the remark this answers is not in the document]
Comment (andrewwa-kainos) [reply src/test/java/uk/nhs/digital/nhsconnect/lab/results/outbound/queue/GpOutboundQueueServiceTest.java:36]: Should we have this service throw an exception if it finds it's processing neither Pathology nor Screening, or map it anyway?

1. Throw exception
2. Map it semantically (i.e., NHS001/NHSACK → "Acknowledgement")
3. Map it generally (i.e., NHS001/NHSACK → "Unknown")

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/MessageHeader.java (hunk 1 above)
Comment (andrewwa-kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/MessageHeader.java]: Yep!

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/outbound/queue/GpOutboundQueueServiceTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewwa-kainos) [reply src/test/java/uk/nhs/digital/nhsconnect/lab/results/outbound/queue/GpOutboundQueueServiceTest.java]: Now throwing an IllegalStateException if NHS001 somehow gets to this outbound service.

**Review by graham-andrew — APPROVED** [review]
Looks good to me

**Review by bartek-sarul — APPROVED** [review]

**Pull request merged**